### DISCUSSION TITLE
Brightness to alpha blend mode

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#2046] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

One popular trick in Mugen is to layer a subtractive sprite under transparent sprites to give them the appearance of 24bit images. This solution is not optimal because you're doubling the number of explods you use and risking desync between them. I get the feeling this could be done directly by code.

### Describe the solution you'd like

"Brightness to alpha" parameter. Probably in the Trans sctrl somehow. What this does is render white as opaque and black as invisible, with gray transparent and so on. But it's different from trans "Add" in that it doesn't actually brighten the background.

The advantage here over just using 24bit images is it would still allow one to change the palette in 8bit images.

**Comment by SuperFromND** [comment]
I wouldn't mind implementing this along with a Multiply blend mode, but last time I asked we'd have to do some degree of refactoring to our blend-mode code.

**Comment by potsmugen** [comment, marked as answer]
Implemented with `SubAdd` type.
